distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/PluginHelper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A helper class for the plugin-cli tasks.
+ */
+public class PluginHelper {
+
+    /**
+     * Verify if a plugin exists with any folder name.
+     * @param pluginPath   the path for the plugins directory.
+     * @param pluginName   the name of the concerned plugin.
+     * @return             the path of the folder if the plugin exists.
+     * @throws IOException if any I/O exception occurs while performing a file operation
+     */
+    public static Path verifyIfPluginExists(Path pluginPath, String pluginName) throws IOException {
```

Comment:
Maybe we can refactor this method to return the name of the plugin instead of the path - something like `getPluginNameFromPluginDescriptor` . This will also make the changes in `InstallPluginCommand `and `RemovePluginCommand `minimal.  You can first check if there is a subdirectory inside `plugins `with the supplied plugin name if not get the name using this method.  What do you think?

Reply:
For the scan, it is kept as a path to be returned since we would have to traverse the paths inside the plugins directory to check for anything existing with the plugin name but installed in a custom folder name or plugin folder name, so it would essentially be the same work to return the path or the name of the folder.